File: sobit_navigation_library/example/move_position_example.cpp
#include <ros/ros.h>
#include <sobit_navigation_library/sobit_navigation_library.hpp>
#include <geometry_msgs/Pose.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "navigation_test");
    ros::NodeHandle nh;
    ros::Rate loop_rate(10);
    SOBITNavigationStack::SOBITNavigationLibrary nav_lib;
    geometry_msgs::Pose target_position;


    target_position.position.x    =  3.0; /* point_x  */
    target_position.position.y    = -2.0; /* point_y  */
    target_position.position.z    =  0.0; /* point_z  */
    target_position.orientation.x =  0.0; /* quater_x */
    target_position.orientation.y =  0.0; /* quater_y */
    target_position.orientation.z =  0.0; /* quater_z */
    target_position.orientation.w =  1.0; /* quater_w */

    /*      move2Position( geometry_msgs::Pose target_position,  frame_name     , is_wait ); */
    nav_lib.move2Position(                     target_position, "base_footprint", false   );
    // nav_lib.move2Position(                     target_position, "map"           , false   );

    while (ros::ok()) {
        if( !nav_lib.exist_goal_ ) {
            break;
        }
        ros::spinOnce();
        loop_rate.sleep();
    }
    ros::spin();
}

File: sobit_navigation_library/example/move_position_example.py
#!/usr/bin/env python3
import rospy
from sobit_navigation_module import SOBITNavigationLibraryPython
import sys


def test():
    rospy.init_node("navigation_test")
    args = sys.argv
    r = rospy.Rate(10) # 10hz
    nav_lib = SOBITNavigationLibraryPython(args[0])

    ####### move2PositionPy( point_x, point_y, point_z, quater_x, quater_y, quater_z, quater_w, frame_name      , is_wait ) #######
    nav_lib.move2PositionPy( 3.0    , -2.0   , 0.0    , 0.0     , 0.0     , 0.0     ,  1.0    , "base_footprint", False   )
    # nav_lib.move2PositionPy( 0.0    ,  0.0   , 0.0    , 0.0     , 0.0     , 0.0     , -1.0    , "map"           , False   )
    while not rospy.is_shutdown():
        if not nav_lib.exist_goal_:
            break
        r.sleep()


if __name__ == "__main__":
    try:
        test()
    except rospy.ROSInterruptException: pass